### PR TITLE
Bump Imath: v3.0.1 -> v3.1.2

### PR DESCRIPTION
**Pull request by twadleigh**
Anticipating a similar bump for OpenEXR.

### CODE REVIEW

File: I/Imath/build_tarballs.jl [hunk 1]
```diff
@@ -31,7 +31,7 @@ platforms = supported_platforms()
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libImath-3_0", :libImath)
+    LibraryProduct("libImath-3_1", :libImath)
```
Comment (giordano) [inline I/Imath/build_tarballs.jl:34]: This means that OpenEXR _**must**_ specify a compat with Imath

Comment (twadleigh) [reply I/Imath/build_tarballs.jl:34]: For my own edification, would there have been more flexibility if I had included the version number in the product symbol?

Comment (twadleigh) [reply I/Imath/build_tarballs.jl:34]: Probably not, since the products would have been disjoint across minor version bumps.

Comment (giordano) [reply I/Imath/build_tarballs.jl:34]: No, I think the name of the product (`:libImath`) without the version number embedded is totally fine.

What I observed is that the filename of the library embeds the version number (and we can do little about that, it comes from the upstream build system), which basically breaks the ABI of the library.

Comment (twadleigh) [reply I/Imath/build_tarballs.jl:34]: Actually, I believe there is a cmake flag to suppress appending the version number to the artifact. Is that something that might be useful to do at some point (maybe not worth it now, but on the next minor version bump)?

**Pull request merged**